engine/src/execution_graph/logic_controllers/BatchProcessing.h

```diff
@@ -133,52 +133,62 @@ class DataSourceSequence {
 		// iterates through files and parses them into columns
 		while(this->provider->has_next()) {
 			// a file handle that we can use in case errors occur to tell the user which file had parsing issues
-			user_readable_file_handles.push_back(this->provider->get_current_user_readable_file_handle());
 			files.push_back(this->provider->get_next());
 		}
 		n_files = files.size();
 		for (size_t index = 0; index < n_files; index++) {
-			ral::io::Schema fileSchema = schema.fileSchema(file_index);
-			std::vector<int> row_groups = fileSchema.get_rowgroup_ids(0);
+			std::vector<cudf::size_type> row_groups = schema.get_rowgroup_ids(index);
 			n_batches += row_groups.size();
 			all_row_groups.push_back(row_groups);
 		}
 
 		is_empty_data_source = (n_files == 0);
+
+		if(is_empty_data_source){
+			n_batches = parser->get_num_partitions();
+		}
 	}
 
 	RecordBatch next() {
 		if (is_empty_data_source) {
-			is_empty_data_source = false;
-			return schema.makeEmptyBlazingTable(projections);
+			if(n_batches==0){
+				is_empty_data_source = false;
+				return schema.makeEmptyBlazingTable(projections);
+			}
+
+			auto ret = loader.load_batch(context.get(), projections, schema, ral::io::data_handle(), cur_file_index, cur_row_group_index);
+			batch_index++;
+			cur_row_group_index++;
+
+			if(batch_index == n_batches){
+				is_empty_data_source = false;
+			}
+
+			return std::move(ret);
 		}
 		
-		//This is just a workaround, mainly for ORC files
-		if(all_row_groups[file_index].size()==1 && all_row_groups[file_index][0]==-1){
-			batch_id = -1; //load all the stripes when can't get the rowgroups size
-		}
+		cudf::size_type row_group_id = all_row_groups[cur_file_index][cur_row_group_index];
 
-		auto ret = loader.load_batch(context.get(), projections, schema, user_readable_file_handles[file_index], files[file_index], file_index, batch_id);
+		auto ret = loader.load_batch(context.get(), projections, schema, files[cur_file_index], cur_file_index, row_group_id);
 		batch_index++;
-		
-		batch_id++;
-		if (batch_id == all_row_groups[file_index].size()) {
+		cur_row_group_index++;
+		if (cur_row_group_index == all_row_groups[cur_file_index].size()) {
```

Comment:
can you explain this csv logic. It seems a little unintuitive. If we must have something like this for now add a comment here explaining why we are doing this and perhaps how we should handle this in the future.

Reply:
I just removed this code. it had no effect and probably depended on something else to work.